### DISCUSSION TITLE
Support for Fedora 39

### DISCUSSION THREAD

**Discussion by harpreetsb** [rabbitmq/rabbitmq-server#10232] [Other] [answered]
### Is your feature request related to a problem? Please describe.

So i recently upgraded to fedora 39, like a month ago.
I am working on personal project and everything worked fine(I did not check my code for AMQP).
But yesterday I was working on the code I found myself into a problem I mentioned here.
https://github.com/php-amqplib/php-amqplib/discussions/1149

### Describe the solution you'd like

Support for Fedora 39

**Comment by michaelklishin** [comment, marked as answer]
There is nothing to support: our RPM package works just fine on Fedora releases between 33 and 39.

See server logs for clues. If this was an upgrade to 3.12, most likely [you haven't had all feature flags enabled before upgrading](https://rabbitmq.com/upgrade.html).

**Comment by michaelklishin** [comment]
From https://github.com/php-amqplib/php-amqplib/discussions/1149:

Our Erlang RPM (both the Fedora 38 and the CentOS Stream 9 versions) supports Fedora 39 just fine.

If this is a development environment, your can remove and purge your installation (that is, its [data directory](https://rabbitmq.com/relocate.html)) then install Erlang 26.2 and RabbitMQ 3.12.11.

If you cannot afford to do that, you must install Erlang 25 and RabbitMQ 3.11.28, then enable **all** feature flags, then upgrade to Erlang 26 and RabbitMQ 3.12.11.

Our [RPM distribution guide](https://rabbitmq.com/install-rpm.html) explains what repositories you should use if you want the latest stuff. Fedora does not provide RabbitMQ 3.12 (or even the latest 3.11) in its repositories (other than Rawhide) but it [does ship with Erlang 26](https://packages.fedoraproject.org/pkgs/erlang/erlang/).

Just nuke the RabbitMQ and Erlang packages installed from Fedora repositories and follow RabbitMQ's RPM guide, please.

**Comment by michaelklishin** [comment]
Apparently the Fedora-packaged RabbitMQ 3.11.10 RPM [assumes that it supports Erlang 26](https://packages.fedoraproject.org/pkgs/rabbitmq-server/rabbitmq-server/fedora-39.html) but that's [not the case](https://rabbitmq.com/which-erlang.html).

RabbitMQ 3.12.x does support Erlang 26 but 3.12.10 is only available via Fedora 39 testing repositories or in Fedora Rawhide.

In any case, a better solution is the same as for other operating systems: [use the repositories maintained by Team RabbitMQ](https://rabbitmq.com/install-rpm.html).
